Fix gerrit upload with mixed explicit and implicit change-ids

HoelzelJon opened the pull request:
This should address the issue described in https://github.com/jj-vcs/jj/issues/8006

This is my first contribution; please let me know if there's anything I've overlooked here. Thanks!

# Checklist

If applicable:

- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (`README.md`, `docs/`, `demos/`) -- N/A
- [ ] I have updated the config schema (`cli/src/config-schema.json`) -- N/A
- [x] I have added/updated tests to cover my changes

yuja commented on cli/tests/test_gerrit_upload.rs:
```diff
@@ -104,11 +107,29 @@ fn test_gerrit_upload_local() {
+    "},
+    )
+    .unwrap();
+    let output = local_dir.run_jj(["describe", "c"]);
```
nit: you can use `describe c -m` instead of the edit script.

HoelzelJon commented:
Also, let me know if it's reasonable that I've added myself to the `paid_contributors.md`. I've implemented this during work hours and with approval from my employer, but since this is a one-off sidequest of mine, I'm not sure if it's best to characterize it as a "compan[y] paying for contributions to Jujutsu". I'm going to leave the addition unless told otherwise, but I just want to make clear what the situation is

HoelzelJon commented on cli/src/commands/gerrit/upload.rs:
```diff
@@ -272,22 +272,20 @@ pub fn cmd_gerrit_upload(
+                if trailers.is_empty() { "\n\n" } else { "\n" },
+                gerrit_change_id
+            )
+        };
```
I've added a pre-commit that splits out two separate tests for all-preset and all-implied Change-Ids, and then in the main commit, I've added a new test case for the relevant mixed case

yuja reviewed (commented):
> Also, let me know if it's reasonable that I've added myself to the `paid_contributors.md`.

I think that's good because the purpose of this list is to spot conflicts of interest.

BTW, I didn't notice CLA issue isn't resolved yet. Some of the contributors wouldn't be allowed to review this PR.

HoelzelJon commented:
> BTW, I didn't notice CLA issue isn't resolved yet. Some of the contributors wouldn't be allowed to review this PR.

Got it, thanks for the context. I'm working on it behind the scenes 👍

yuja commented on docs/paid_contributors.md:
```diff
@@ -54,3 +54,6 @@ See [contribution docs](contributing.md#code-reviews) for details on this policy
 * zygoloid
+
+## IMC Trading
+* HoelzelJon
```
nit: insert blank line after heading

HoelzelJon commented:
looks like the merge queue build failed on a timeout. I'm not seeing any indication that that's related to the changes in this commit, so I'm gonna try merging again
